app/services/FinancialDataService.scala

```diff
@@ -30,49 +30,34 @@ import scala.concurrent.{ExecutionContext, Future}
 class FinancialDataService @Inject()(
                                       financialDataConnector: FinancialDataConnector,
                                       vatReturnService: VatReturnService,
-                                      periodService: PeriodService,
-                                      clock: Clock
+                                      periodService: PeriodService
                                     )(implicit ec: ExecutionContext) extends Logging {
 
   def getCharge(vrn: Vrn, period: Period): Future[Option[Charge]] = {
-    getFinancialData(vrn, period.firstDay, period.lastDay).map { maybeFinancialDataResponse =>
-      maybeFinancialDataResponse.flatMap {
-        financialDataResponse =>
-          financialDataResponse.financialTransactions.flatMap {
-            transactions =>
-              getChargeForPeriod(period, transactions)
+    getFinancialDataForDateRange(vrn, period.firstDay, period.lastDay).map {
+      _.flatMap {
+        _.financialTransactions.flatMap {
+          transactions =>
+            getChargeForPeriod(period, transactions)
           }
-      }
+        }
     }
   }
 
   def getVatReturnWithFinancialData(vrn: Vrn, commencementDate: LocalDate): Future[Seq[VatReturnWithFinancialData]] = {
-
-    val getFinancialDataResponse = financialDataConnector
-      .getFinancialData(vrn,
-        FinancialDataQueryParameters(
-          fromDate = Some(commencementDate),
-          toDate = Some(LocalDate.now(clock))
-        )).map {
-      case Right(maybeFinancialData) => maybeFinancialData
-      case Left(_) => None
-    }.recover {
-      case _: Exception =>
-        None
-    }
-
     for {
       vatReturns <- vatReturnService.get(vrn)
-      maybeFinancialDataResponse <- getFinancialDataResponse
+      maybeFinancialDataResponse <- getFinancialData(vrn, commencementDate).recover {
+        case _: Exception => None
```

Comment:
Can we add a log line here with the exception passed in?

Reply:
done now